src/app/wallets/get-last-on-chain-address.ts

```diff
@@ -0,0 +1,13 @@
+import { WalletOnChainAddressesRepository } from "@services/mongoose"
+import { createOnChainAddress } from "./create-on-chain-address"
+
+export const getLastOnChainAddress = async (
+  walletId: WalletId,
+): Promise<OnChainAddress | ApplicationError> => {
+  const onChainAddressesRepo = WalletOnChainAddressesRepository()
+  const lastOnChainAddress = await onChainAddressesRepo.findLastByWalletId(walletId)
+
+  if (lastOnChainAddress instanceof Error) return createOnChainAddress(walletId)
```

Comment:
This should only be on `CouldNotFindError` not on _every_ `Error`.

Reply:
fixed